app/services/foreman/plugin.rb

```diff
@@ -351,6 +351,12 @@ def parameter_filters(klass)
       @parameter_filters.fetch(klass.is_a?(Class) ? klass.name : klass, [])
     end
 
+    def hosts_controller_action_scope(action, &block)
+      # Add scope for both API and UI controller.
+      HostsController.add_scope_for(action, block)
```

Comment:
I think, you need to pass the block like so (note the extra ampersand):

```ruby
HostsController.add_scope_for(action, &block)
```

Same below.